File: src/app/models/job.ts
export class Job {
private _description: string;
private _createDate: string;
private _jobType: string;
private _alarmedUnits: string[];
private _reporter: string;
private _adress: string;
private _uuid: string;
private _running: boolean;


  constructor(description: string, createDate: string, jobType: string, alarmedUnits: string[], reporter: string, adress: string, uuid: string, running: boolean) {
    this._description = description;
    this._createDate = createDate;
    this._jobType = jobType;
    this._alarmedUnits = alarmedUnits;
    this._reporter = reporter;
    this._adress = adress;
    this._uuid = uuid;
    this._running = false;
  }


  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get createDate(): string {
    return this._createDate;
  }

  set createDate(value: string) {
    this._createDate = value;
  }

  get jobType(): string {
    return this._jobType;
  }

  set jobType(value: string) {
    this._jobType = value;
  }

  get alarmedUnits(): string[] {
    return this._alarmedUnits;
  }

  set alarmedUnits(value: string[]) {
    this._alarmedUnits = value;
  }

  get reporter(): string {
    return this._reporter;
  }

  set reporter(value: string) {
    this._reporter = value;
  }

  get adress(): string {
    return this._adress;
  }

  set adress(value: string) {
    this._adress = value;
  }

  get uuid(): string {
    return this._uuid;
  }

  set uuid(value: string) {
    this._uuid = value;
  }


  public get running(): boolean {
    return this._running;
  }

 public set running(value: boolean) {
    this._running = value;
  }
}

File: alarmmonitor-server/src/main/java/de/alarmmonitor/server/models/Job.java
package de.alarmmonitor.server.models;

import de.alarmmonitor.server.enums.Car;
import de.alarmmonitor.server.enums.JobType;

import java.util.Date;
import java.util.List;
import java.util.UUID;

public class Job {
    private String description;
    private Date createDate;
    private JobType jobType;
    private List<Car> alarmedUnits;
    private String reporter;
    private String adress;
    private String uuid;
    boolean running;

    public Job(String description, Date createDate, JobType jobType, List<Car> alarmedUnits, String reporter, String adress) {
        this.description = description;
        this.createDate = createDate;
        this.jobType = jobType;
        this.alarmedUnits = alarmedUnits;
        this.reporter = reporter;
        this.adress = adress;
        generateJobId();
    }

    private void generateJobId() {
        this.uuid = UUID.randomUUID().toString();
        this.running = false;
    }

    @Override
    public String toString() {
        return this.jobType.getShortcut() + ": " + this.adress;
    }

    public String carsAsString() {
        StringBuilder b = new StringBuilder();
        for (Car c : alarmedUnits
                ) {
            b.append(c.getShortcut());
            b.append(" ");
        }
        return b.toString();
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getCreateDate() {
        return createDate;
    }

    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }

    public JobType getJobType() {
        return jobType;
    }

    public void setJobType(JobType jobType) {
        this.jobType = jobType;
    }

    public List<Car> getAlarmedUnits() {
        return alarmedUnits;
    }

    public void setAlarmedUnits(List<Car> alarmedUnits) {
        this.alarmedUnits = alarmedUnits;
    }

    public String getReporter() {
        return reporter;
    }

    public void setReporter(String reporter) {
        this.reporter = reporter;
    }

    public String getAdress() {
        return adress;
    }

    public void setAdress(String adress) {
        this.adress = adress;
    }

    public String getUuid() {
        return uuid;
    }

    public void setRunning(boolean running) {
        this.running = running;
    }

    public boolean isRunning() {
        return running;
    }
}
